### PR TITLE
Adds topic query functionality to GET /api/articles with error handling and associated tests

### PR DESCRIPTION
**Pull request by Radishetta**
Adds topic query functionality to GET /api/articles
Updates app.test.js
Updates app.js
Updates articles.controller.js
Updates articles.model.js
Updates endpoints.json

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -126,6 +126,42 @@ describe("GET /api/articles", () => {
...
+
+        expect(articles.length).toBe(12);
+        expect(Array.isArray(articles)).toBe(true);
+
+        articles.forEach((article) => {
+          expect(article).toMatchObject({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            topic: "mitch",
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:145]: Lovely!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -126,6 +126,42 @@ describe("GET /api/articles", () => {
   });
 });
 
+describe("GET /api/articles - topic query", () => {
+  it("200: responds with articles associated to the given topic", () => {
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:130]: Really nice test!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -126,6 +126,42 @@ describe("GET /api/articles", () => {
   });
 });
 
+describe("GET /api/articles - topic query", () => {
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:129]: You need another test to check we get an empty array back when there are no articles for a given topic that does exist.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -126,6 +126,42 @@ describe("GET /api/articles", () => {
...
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number),
+          });
+        });
+      });
+  });
+
+  it("404: ERROR - responds with an error message when passed and inexistent topic", () => {
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:155]: Good test

**Review by nc-aimee-muter — COMMENTED** [review]
Looking good but it is missing one test
